### ISSUE TITLE
CGAL kernel has no support IFCSWEPTDISKSOLID (ReinforcingBar)

### ISSUE THREAD

**Issue by csritter** [IfcOpenShell/IfcOpenShell#5688]
Trying to convert an ifc file with ReinforcingBar and it shows me this issue:

`[Error] {3LONVhcYDApOkVkNgy78bP} No conversion with upgrade for 21`

Opencascade works fine.

I will send the file on private email and forms.

**Comment by aothms** [comment]
Yes, this is to be expected. CGAL is an awesome library, but it does not contain the same wide variety of CAD primitives that opencascade supports. So some of these sweeps are currently simply not supported. We can add support for them manually, but it's not highest priority. Recommended approach for using cgal is with `hybrid-cgal-simple-opencascade` so that opencascade is used as a fallback when cgal does not support something.

These kind of things are potentially fun to work on though. I'll mark it as a good first issue for someone with an interest in computational geometry :)

**Comment by csritter** [comment]
Hi @aothms, thanks for the answer.
One more question... Can I use `hybrid-cgal-opencascade` instead of `hybrid-cgal-simple-opencascade`?
Or is there something more performative on `hybrid-cgal-simple-opencascade`?

**Comment by aothms** [comment]
You can. The implications of that are that you'd also use CGAL for boolean operations. Which should be very reliable in general, but without tolerance. We have three ways for applying tolerance (a) minkowski to dilate (b) snap operands as halfspaces (c) snap vertices. Only (a) generally works, but is quite slow.
